src/abstractdataframe/selection.jl

```diff
@@ -0,0 +1,357 @@
+normalize_selection(idx::AbstractIndex, sel) = idx[sel]
+normalize_selection(idx::AbstractIndex, sel::Pair{<:ColumnIndex,<:Pair{<:Function,Symbol}}) =
+    idx[first(sel)] => last(sel)
+normalize_selection(idx::AbstractIndex, sel::Pair{<:ColumnIndex, <:Symbol}) =
+    idx[first(sel)] => last(sel)
+
+function normalize_selection(idx::AbstractIndex, sel::Pair{<:ColumnIndex, <:Function})
```

Comment:
This could be useful to allow converting columns:
```suggestion
function normalize_selection(idx::AbstractIndex, sel::Pair{<:ColumnIndex, <:Union{Function,Type}})
```
Same elsewhere.

Reply:
OK, I will add it (I wanted to be restrictive initially). But I will just use `Base.Callable` in the code.